include/systems/eigen_system.h

```diff
@@ -180,56 +170,212 @@ class EigenSystem : public System
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if !_use_shell_matrices.
+   */
+  SparseMatrix<Number> & get_matrix_A();
+
+  /**
+   * \returns A const reference to the system matrix used for generalized eigenvalue problems
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if !_use_shell_matrices and generalized().
+   */
+  const SparseMatrix<Number> & get_matrix_B() const;
+
+  /**
+   * \returns A const reference to the system matrix used for generalized eigenvalue problems
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if !_use_shell_matrices and generalized().
+   */
+  SparseMatrix<Number> & get_matrix_B();
+
+  /**
+   * \returns A const reference to the preconditioning matrix.
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if !_use_shell_matrices.
+   */
+  const SparseMatrix<Number> & get_precond_matrix() const;
+
+  /**
+   * \returns A reference to the preconditioning matrix.
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if !_use_shell_matrices.
+   */
+  SparseMatrix<Number> & get_precond_matrix();
+
+  /**
+   * \returns A const reference to the system shell matrix used for standard eigenvalue problems.
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if _use_shell_matrices.
+   */
+  const ShellMatrix<Number> & get_shell_matrix_A() const;
+
+  /**
+   * \returns A reference to the system shell matrix used for standard eigenvalue problems.
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if _use_shell_matrices.
+   */
+  ShellMatrix<Number> & get_shell_matrix_A();
+
+  /**
+   * \returns A const reference to the system shell matrix used for generalized eigenvalue problems.
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if _use_shell_matrices and generalized().
+   */
+  const ShellMatrix<Number> & get_shell_matrix_B() const;
+
+  /**
+   * \returns A reference to the system shell matrix used for generalized eigenvalue problems.
+   *
+   * This matrix should only be available (and therefore this should only be called)
+   * if _use_shell_matrices and generalized().
+   */
+  ShellMatrix<Number> & get_shell_matrix_B();
+
+  /**
+   * \returns A const reference to the system shell matrix used for preconditioning.
+   */
+  const ShellMatrix<Number> & get_shell_precond_matrix() const;
+
+  /**
+   * \returns A reference to the system shell matrix used for preconditioning.
+   */
+  ShellMatrix<Number> & get_shell_precond_matrix();
+
+  /**
+   * \returns A const reference to the EigenSolver.
+   */
+  inline const EigenSolver<Number> & get_eigen_solver() const
```

Comment:
Thanks for moving the inlined functions to the C file! Looks like you got all of them except these two `get_eigen_solver()`  calls.

Reply:
Whoops - I saw those, didn't remember putting them in, and didn't check that it wasn't me that put them in.

Testing wasn't done anyway, just moved them as well. I'm surprised that another approval isn't needed - is there a github setting that disables automatic approval removal after a new push?

Reply:
Yeah, looks like [it's an option](https://stackoverflow.com/questions/41824374/how-to-un-approve-github-review-after-new-commits-automatically).